module/Activity/src/Activity/Service/Activity.php

```diff
@@ -110,12 +111,30 @@ public function createActivity(array $params)
         // Find the creator
         $user = $this->getServiceManager()->get('user_role');
         if ($user === 'guest') {
-            throw new \InvalidArgumentException('Guests can not create activities');
+            throw new NotAllowedException('Guests can not create activities');
         }
-        $params['creator'] = $user;
-
         $activity = new ActivityModel();
-        $activity->create($params);
+        $activity->setBeginTime(new \DateTime($params['beginTime']));
+        $activity->setEndTime(new \DateTime($params['endTime']));
+        $activity->setName($params['name']);
+        $activity->setNameEn($params['name_en']);
+        $activity->setLocation($params['location']);
+        $activity->setLocationEn($params['location_en']);
+
+        if (!$params['costs_unknown']) {
+            $activity->setCosts($params['costs']);
+            $activity->setCostsEn($params['costs_en']);
+        }
```

Comment:
I don't really see much magic in the forms actually. If you use `bind()` (which is the most magic there is), it will be filled by a hydrator (which you have to set yourself).